File: debugging_polymorphism.py
class Pony:
    name: str

    legCount: int = 4

    def __init__(self, name : str):
        """ Create a new regular Pony by name """
        self.name = name

class Pegasus(Pony):
    wingCount: int = 2

class Unicorn(Pony):
    hornCount: int = 1

aj = Pony('Applejack')
fs = Pegasus('Fluttershy')
ts = Unicorn('Twilight Sparcle')

PonyList = list[Pony]

ponyList = PonyList([aj, fs, ts])

for pony in ponyList:
  print(pony.name)

File: debugging_polymorphism.js
class Pony {
    constructor(name, legCount) {
        this.name = name;
        this.legCount = legCount;
    }
}


class Pegasus extends Pony {
    constructor(name, legCount) {
        super(name, legCount);
        this.wingCount = 2;
    }
}


class Unicorn extends Pony {
    constructor(name, legCount) {
        super(name, legCount);
        this.hornCount = 1;
    }
}

aj = new Pony('Applejack');
fs = new Pegasus('Fluttershy');
ts = new Unicorn('Twilight Sparcle');

 

ponyList = [aj, fs, ts];

ponyList.forEach(pony => {

    
    console.log(pony.wingCount)

});

